### PR TITLE
[QPG6100] Adding implementation of BleManager

### LINKED ISSUE #3112: [QPG6100] Add BLE connectivity
#### Problem
QPG6100 example is missing BLE Manager integration.
This blocks any BLE based provisioning application to be handled.

 #### Proposed Solution
* Add BLE Manager implementation based on Qorvo SDK code.
* Enable CHIPoBLE advertising + connection possibility on example application.

### PR DESCRIPTION
**Pull request by tima-q**
#### Problem
BLEManager for QPG6100 curretnly only stubbed.
PR enables CHIPoBLE functionality on QPG6100 - Issue #3112

Example application will start advertisement and accept connections.

 #### Summary of Changes

* Adding implementation of BleManager for QPG6100
* Updating qpg_sdk submodule to supply necessary updates
* README update

### CODE REVIEW

File: src/platform/qpg6100/BLEManagerImpl.cpp [hunk 1]
```diff
@@ -39,6 +39,15 @@ namespace Internal {
 
 namespace {
 
+const uint16_t UUID16_CHIPoBLEService = 0xFEAF;
+// const ble_uuid_t UUID_CHIPoBLEService = { UUID16_CHIPoBLEService, BLE_UUID_TYPE_BLE };
```
Comment (andy31415) [inline src/platform/qpg6100/BLEManagerImpl.cpp:43]: Please remove commented out code.

File: src/platform/qpg6100/BLEManagerImpl.cpp [hunk 2]
```diff
@@ -120,53 +132,37 @@ CHIP_ERROR BLEManagerImpl::_SetFastAdvertisingEnabled(bool val)
...
-    // TODO
-    // err = qvCHIP_BleSetDeviceName(devName);
-    goto exit;
+    if (mServiceMode == ConnectivityManager::kCHIPoBLEServiceMode_NotSupported)
+    {
+        return CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE;
+    }
+    if (devName != NULL && devName[0] != 0)
+    {
+        err = qvCHIP_BleSetDeviceName(devName);
```
Comment (andy31415) [inline src/platform/qpg6100/BLEManagerImpl.cpp:152]: should this err (if it happens) be returned? Alternatively if this is not success, should we log it?

it seems like there is some double-return happening below.

File: src/platform/qpg6100/BLEManagerImpl.cpp [hunk 3]
```diff
@@ -405,67 +389,270 @@ void BLEManagerImpl::DriveBLEState(intptr_t arg)
...
+}
 
-    // Update advertising state on connection
-    SetFlag(mFlags, kFlag_AdvertisingRefreshNeeded);
-    PlatformMgr().ScheduleWork(DriveBLEState, 0);
+void BLEManagerImpl::HandleRXCharWrite(uint16_t connId, uint16_t handle, uint8_t operation, uint16_t offset, uint16_t len,
+                                       uint8_t * pValue, qvCHIP_Ble_Attr_t * pAttr)
+{
+    CHIP_ERROR err     = CHIP_NO_ERROR;
+    PacketBuffer * buf = NULL;
```
Comment (andy31415) [inline src/platform/qpg6100/BLEManagerImpl.cpp:407]: nit: I believe we currently mostly use `nullptr` in CHIP

File: src/platform/qpg6100/BLEManagerImpl.cpp (hunk 2 above)
Comment (tima-q) [reply src/platform/qpg6100/BLEManagerImpl.cpp:152]: Correct - Remnant return from 'goto exit:' removal

File: src/platform/qpg6100/BLEManagerImpl.cpp (hunk 3 above)
Comment (tima-q) [reply src/platform/qpg6100/BLEManagerImpl.cpp:407]: agreed.
Note - ironically - only search hits on NULL in https://github.com/project-chip/connectedhomeip/blob/master/docs/style/coding/CODING_STYLE_GUIDE.adoc

**Review by bzbarsky-apple — APPROVED** [review]

File: src/platform/qpg6100/BLEManagerImpl.cpp (hunk 3 above)
Comment (andy31415) [reply src/platform/qpg6100/BLEManagerImpl.cpp]: That is hilarious :rofl:

**Review by andy31415 — APPROVED** [review]

**Review by rwalker-apple — APPROVED** [review]

**Review by mspang — APPROVED** [review]

**Review by BroderickCarlin — APPROVED** [review]

**Pull request merged**